src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,172 @@
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM shopping_carts WHERE shopping_cart_id = ?;";
+        String deleteFromShoppingCartProductsQuery =
+                "DELETE FROM shopping_cart_products WHERE shopping_cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement deleteFromShoppingCartProductsStatement =
+                    connection.prepareStatement(deleteFromShoppingCartProductsQuery);
+            deleteFromShoppingCartProductsStatement.setLong(1, id);
+            deleteFromShoppingCartProductsStatement.executeUpdate();
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete shopping cart with id: " + id, e);
+        }
+    }
+
+    private void setShoppingCartProducts(ShoppingCart shoppingCart, Connection connection)
+            throws SQLException {
+        String setOrderProductsQuery =
+                "INSERT INTO shopping_cart_products (shopping_cart_products.shopping_cart_id"
+                        + ", shopping_cart_products.product_id) "
+                        + "VALUES (?, ?)";
+        PreparedStatement preparedStatement =
+                connection.prepareStatement(setOrderProductsQuery);
+        for (Product product : shoppingCart.getProducts()) {
+            preparedStatement.setLong(1, shoppingCart.getId());
+            preparedStatement.setLong(2, product.getId());
+            preparedStatement.executeUpdate();
+        }
+    }
+
+    private List<ShoppingCart> getShoppingCartsFromResultSet(ResultSet resultSet)
+            throws SQLException {
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        List<Product> products = new ArrayList<>();
+        while (resultSet.next()) {
+            if (shoppingCarts.isEmpty()
+                    || !shoppingCarts.get(shoppingCarts.size() - 1).getId()
+                    .equals(resultSet.getLong("shopping_cart_id"))) {
+                products = new ArrayList<>();
+                ShoppingCart shoppingCart = new ShoppingCart(
+                        products,
+                        resultSet.getLong("shopping_cart_user_id")
+                );
+                shoppingCart.setId(resultSet.getLong("shopping_cart_id"));
+                shoppingCarts.add(shoppingCart);
+            }
+            if (resultSet.getString("product_name") != null) {
+                Product product = new Product(
+                        resultSet.getString("product_name"),
+                        resultSet.getBigDecimal("price")
+                );
+                product.setId(resultSet.getLong("product_id"));
+                products.add(product);
+            }
+        }
+        return shoppingCarts;
```

Comment:
let's make loop which will return only one shoppingCart and when you will use getAll() then you can call this method in loop there

Reply:
got it )